### PR TITLE
[RFC] Add multilib sni-qt

### PR DESCRIPTION
**Pull request by karolherbst**
This pull request adds multilib support for sni-qt fixing these gentoo bugs: https://bugs.gentoo.org/show_bug.cgi?id=516488 https://bugs.gentoo.org/show_bug.cgi?id=530154

The commits basically copy ebuilds from upstream gentoo and then add multilib support in needed packages:
* qjson
* libdbusmenu-qt
* sni-qt

I've tested it with skype inside plasma5 and the systray icons shows up.

### CODE REVIEW

**Comment by johu** [comment]
Please cleanup cvs header.

**Comment by kensington** [comment]
@mgorny do you have any multilib comments? I'm keen to get this to the tree.

File: dev-libs/libdbusmenu-qt/files/libdbusmenu-qt-0.9.3_pre20140619-optionaltests.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
++
++option(BUILD_TESTS "Build tests for qt4" ON)
+ if (NOT USE_QT5) # TODO port tests to Qt5
+-add_subdirectory(tests)
++	if(BUILD_TESTS)
++		add_subdirectory(tests)
++	endif(BUILD_TESTS)
+ endif()
+ add_subdirectory(tools)
+
```
Comment (mgorny) [inline dev-libs/libdbusmenu-qt/files/libdbusmenu-qt-0.9.3_pre20140619-optionaltests.patch]: Looks like stray empty line :P.

File: dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		$(cmake-utils_use_with doc)
+		-DUSE_QT5=ON
+	)
+	cmake-utils_src_configure
+}
+
+src_configure() {
+	myconfigure() {
+		if [[ ${MULTIBUILD_VARIANT} = qt4 ]] ; then
+			multilib-minimal_src_configure
```
Comment (mgorny) [inline dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild]: It ain't pretty when you call `multilib_minimal_*` once, and `cmake-multilib_*` the other time :).

File: dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	multibuild_foreach_variant mytest
+}
+
+multilib_src_test() {
+	local builddir=${BUILD_DIR}
+
+	BUILD_DIR=${BUILD_DIR}/tests \
+		VIRTUALX_COMMAND=cmake-utils_src_test virtualmake
+
+	BUILD_DIR=${builddir}
```
Comment (mgorny) [inline dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild]: What's that for? Sub-function calls don't override vars in bash, so your `BUILD_DIR=... virtualmake` is safe.

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+inherit base cmake-multilib
```
Comment (mgorny) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: `base` is deprecated to the level of QA ban!

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+	cmake-utils_src_configure
+}
+
+multilib_src_install_all() {
+	if use doc; then
+		cd doc
+		doxygen Doxyfile || die "Generating documentation failed"
+		HTML_DOCS=( doc/html/ )
+		base_src_install_docs
```
Comment (mgorny) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: `einstalldocs`?

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		-DQT_QMAKE_EXECUTABLE=/usr/$(get_libdir)/qt4/bin/qmake
+	)
+
+	cmake-utils_src_configure
+}
+
+multilib_src_install_all() {
+	if use doc; then
+		cd doc
+		doxygen Doxyfile || die "Generating documentation failed"
```
Comment (mgorny) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: This belongs in `src_compile`.

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		$(cmake-utils_use test QJSON_BUILD_TESTS)
+		-DQT_QMAKE_EXECUTABLE=/usr/$(get_libdir)/qt4/bin/qmake
+	)
+
+	cmake-utils_src_configure
+}
+
+multilib_src_install_all() {
+	if use doc; then
+		cd doc
```
Comment (mgorny) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: `|| die`

File: dev-libs/libdbusmenu-qt/files/libdbusmenu-qt-0.9.3_pre20140619-optionaltests.patch (hunk 1 above)
Comment (karolherbst) [inline dev-libs/libdbusmenu-qt/files/libdbusmenu-qt-0.9.3_pre20140619-optionaltests.patch]: yeah, but this is the patch already in the portage main tree, so I just left it as it is

File: dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild (hunk 2 above)
Comment (karolherbst) [inline dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild]: I know, but this makes things easier and more readable. Ideally stuff like that could be in some sort of eclass.

File: dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild (hunk 3 above)
Comment (karolherbst) [inline dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild]: taken from gentoo-upstream, don't want to change it inside this changes. Cleanups should be seperated I guess.

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild (hunk 4 above)
Comment (karolherbst) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: repoman didn't complain ;)

File: dev-libs/qjson/qjson-0.8.1-r1.ebuild (hunk 6 above)
Comment (karolherbst) [inline dev-libs/qjson/qjson-0.8.1-r1.ebuild]: is there a nice way to handle this with multilib?

**Comment by karolherbst** [comment]
so I took care about the comments, allthough there are two left, but these shouldn't be taken care of there I guess.

**Comment by kensington** [comment]
Thanks! Merged to main tree.

**Pull request closed without merging**